### PR TITLE
Tutorial 02

### PR DESCRIPTION
**Pull request by nick-harder**
This commit adds the new example. 
It also edits multiple docstring transitioning to the google style docstings. The whole common folder has been edited.
Tests have been also slightly modified, purely cosmetical changes where all marketconfigs have now parameter_name=parameter for easier reading.

### CODE REVIEW

**Review by kim-mskw — CHANGES_REQUESTED** [review]
Love the tutorials, only have minor remarks/questions :)

File: assume/common/market_objects.py [hunk 1]
```diff
@@ -71,40 +62,29 @@ class MarketProduct:
...
 
-    duration: rd | rr.rrule  # quarter-hourly, half-hourly, hourly, 4hourly, daily, weekly, monthly, quarter-yearly, yearly
-    count: int  # how many future durations can be traded, must be >= 1
-    # count can also be given as a rrule with until
-    first_delivery: rd = (
-        rd()
-    )  # when does the first delivery begin, in relation to market start
-    # this should be a multiple of duration
+    duration: rd | rr.rrule
+    count: int
```
[the remark this answers is not in the document]
Comment (kim-mskw) [reply assume/common/market_objects.py:74]: Was that liek a concise choice of having less comennts, or was this just with regard to these comments?

File: assume/common/scenario_loader.py [hunk 2]
```diff
@@ -464,28 +479,91 @@ async def load_scenario_folder_async(
...
+            scenario="scenario_name",
+            study_case="study_case_name",
+            perform_learning=True,
+            perform_evaluation=False,
+            episode=1,
+            eval_episode=1,
+            trained_actors_path="",
+        )
+
+    Notes:
```
[the remark this answers is not in the document]
Comment (nick-harder) [reply assume/common/scenario_loader.py:525]: yes, the idea is to have a description but extra contents go here

File: assume/world.py [hunk 3]
```diff
@@ -133,34 +131,30 @@ async def setup(
         save_frequency_hours: int = 24,
         bidding_params: dict = {},
         learning_config: LearningConfig = {},
-        forecaster: Forecaster = None,
-        manager_address=None,
-        **kwargs,
-    ):
+        forecaster: Optional[Forecaster] = None,
+        manager_address: Optional[Any] = None,
```
[the remark this answers is not in the document]
Comment (maurerle) [reply assume/world.py:135]: Where is this still missing?

File: examples/notebooks/04_Reinforcement_learning_example.ipynb [hunk 4]
```diff
@@ -6,7 +6,7 @@
     "id": "4JeBorbE6FYr"
    },
    "source": [
-    "# Reinforcement learning example\n",
+    "# 4. Reinforcement learning example\n",
```
[the remark this answers is not in the document]
Comment (maurerle) [reply examples/notebooks/04_Reinforcement_learning_example.ipynb]: solved

File: assume/common/market_objects.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (maurerle) [reply assume/common/market_objects.py:74]: The comments are added to the docs above, so its fine

**Review by maurerle — APPROVED** [review]
Looks good

**Comment by maurerle** [comment]
It does not look good at all.
The new doc style is so broken.

https://assume.readthedocs.io/en/latest/assume.common.html#assume.common.forecasts.CsvForecaster.calculate_residual_demand_forecast

**Comment by maurerle** [comment]
This is how it should look like:
image

This is how the GPT-docs style shows up:
image

We can merge this as usual and fix it afterwards.
I don't think that there is an automated way to parse the GPT docs suggestions and fix them.

File: assume/common/scenario_loader.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (kim-mskw) [reply assume/common/scenario_loader.py:525]: Yeah I was just wondering about the label "notes". But then we do it this way.

[the remark this answers is not in the document]
Comment (kim-mskw) [reply assume/common/scenario_loader.py:525]: I documented that as well

File: assume/world.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (kim-mskw) [reply assume/world.py:135]: Done ;)

**Review by kim-mskw — APPROVED** [review]

**Pull request merged**